src/Common/src/System/Net/CookieParser.cs

```diff
@@ -542,6 +543,18 @@ internal string GetString()
             return _tokenizer.GetCookieString();
         }
 
+        private static void SetCookieName(Cookie cookie, string name)
+        {
+            try
+            {
+                cookie.Name = name;
+            }
+            catch (CookieException)
+            {
+                Debug.Assert(cookie.Name == string.Empty);
+            }
```

Comment:
You're right, `Cookie.set_Name` and `Cookie.InternalSetName` have different properties when passing `string.Empty` to them.

Cookie:

```
public string Name
{
    get
    {
        return _name;
    }
    set
    {
        if (String.IsNullOrEmpty(value) || !InternalSetName(value))
        {
            throw new CookieException(SR.Format(SR.net_cookie_attribute, "Name", value == null ? "<null>" : value));
        }
    }
}

internal bool InternalSetName(string value)
{
    if (String.IsNullOrEmpty(value) || value[0] == '$' || value.IndexOfAny(ReservedToName) != -1)
    {
        _name = string.Empty;
        return false;
    }
    _name = value;
    return true;
}
```

Notice that if we pass `string.Empty` to `set_Name`, `Cookie.Name` stays the same as it was before, but if we pass `string.Empty` to `InternalSetName`, `Cookie.Name` is set to `string.Empty`.

I've worked around this by doing reflection. This is an older commit